drivers/rtc/rtc_rv8263.c

```diff
@@ -0,0 +1,784 @@
+	regs[1] = bin2bcd(timeptr->tm_min) & MINUTES_BITS;
+	regs[2] = bin2bcd(timeptr->tm_hour) & HOURS_BITS;
+	regs[3] = bin2bcd(timeptr->tm_mday) & DATE_BITS;
+	regs[4] = bin2bcd(timeptr->tm_wday) & WEEKDAY_BITS;
+	regs[5] = bin2bcd(timeptr->tm_mon) & MONTHS_BITS;
+	regs[6] = bin2bcd(timeptr->tm_year - RV8263_YEAR_OFFSET) & YEAR_BITS;
+
+	return i2c_burst_write_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS, regs, sizeof(regs));
+}
+
+static int rv8263c8_time_get(const struct device *dev, struct rtc_time *timeptr)
+{
+	int err;
+	uint8_t regs[7];
+	const struct rv8263c8_config *config = dev->config;
+
+	if (timeptr == NULL) {
+		return -EINVAL;
+	}
+
+	err = i2c_burst_read_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS, regs, sizeof(regs));
+	if (err < 0) {
+		return err;
+	}
+
+	/* Return an error when the oscillator is stopped. */
+	if (regs[0] & RV8263_BM_OS) {
+		return -ECANCELED;
+	}
+
+	timeptr->tm_sec = bcd2bin(regs[0] & SECONDS_BITS);
+	timeptr->tm_min = bcd2bin(regs[1] & MINUTES_BITS);
+	timeptr->tm_hour = bcd2bin(regs[2] & HOURS_BITS);
+	timeptr->tm_mday = bcd2bin(regs[3] & DATE_BITS);
+	timeptr->tm_wday = bcd2bin(regs[4] & WEEKDAY_BITS);
+	timeptr->tm_mon = bcd2bin(regs[5] & MONTHS_BITS);
+	timeptr->tm_year = bcd2bin(regs[6] & YEAR_BITS) + RV8263_YEAR_OFFSET;
+
+	/* Unused. */
+	timeptr->tm_nsec = 0;
+	timeptr->tm_isdst = -1;
+	timeptr->tm_yday = -1;
+
+	/* Validate the chip in 24hr mode. */
+	if (regs[2] & VALIDATE_24HR) {
+		return -ENODATA;
+	}
+
+	LOG_DBG("Get time: year = %u, mon = %u, mday = %u, wday = %u, hour = %u, min = %u, sec = "
+		"%u",
+		timeptr->tm_year, timeptr->tm_mon, timeptr->tm_mday, timeptr->tm_wday,
+		timeptr->tm_hour, timeptr->tm_min, timeptr->tm_sec);
+
+	return 0;
+}
+
+static int rv8263c8_init(const struct device *dev)
+{
+	int err;
+	int temp;
+	struct rv8263c8_data *data = dev->data;
+	const struct rv8263c8_config *config = dev->config;
+
+#if (CONFIG_RTC_ALARM || CONFIG_RTC_UPDATE) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	if (config->int_gpio.port == NULL) {
+		return -EINVAL;
+	}
+#endif
+
+	if (!i2c_is_ready_dt(&config->i2c_bus)) {
+		LOG_ERR("I2C bus not ready!");
+		return -ENODEV;
+	}
+
+	k_sem_init(&data->lock, 1, 1);
+
+	err = rv8263c8_update_disable_timer(dev);
+	if (err < 0) {
+		LOG_ERR("Error while disabling the timer! Error: %i", err);
+		return err;
+	}
+
+	err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_1,
+				    RV8263C8_BM_24H_MODE_DISABLE | RV8263C8_BM_CLOCK_ENABLE);
+	if (err < 0) {
+		LOG_ERR("Error while writing CONTROL_1! Error: %i", err);
+		return err;
+	}
+
+	temp = 0x00;
+	if (config->clkout == 0) {
+		temp = 0x07;
+	} else if (config->clkout == 1) {
+		temp = 0x06;
+	} else if (config->clkout == 1024) {
+		temp = 0x05;
+	} else if (config->clkout == 2048) {
+		temp = 0x04;
+	} else if (config->clkout == 4096) {
+		temp = 0x03;
+	} else if (config->clkout == 8192) {
+		temp = 0x02;
+	} else if (config->clkout == 16384) {
+		temp = 0x01;
+	}
```

Comment:
As the enumeration list exactly matches the required register value, you could use `DT_INST_ENUM_IDX` to avoid having to calculate the value at runtime :)

Reply:
2 kHz was also missing in the DTS file.